### ISSUE TITLE
npm install returns immediately and without any errors when network unavailable

### ISSUE THREAD

**Issue by psnider** [npm/npm#10064]
npm version 2.11.3, running on RedHat EL7

`npm install` returned without any output, and with a process exit status code of zero,
when we were running it on a machine that has a network card problem 
that results in intermittent or unavailable connections.

There were definitely packages that had not yet been installed.

I think you have a bug in which npm fails early due to the network being unavailable,
but then doesn't take any action to communicate the failure.

**Comment by KenanY** [comment]
@psnider Does this also occur with the latest stable release of `npm`?

**Comment by KenanY** [comment]
@psnider It has been several months now with no reply from you... are you still seeing this issue?

**Comment by psnider** [comment]
Just retested using npm 3.3.9 (on a recent MacBook on El Capitan)

I turned off my wifi, (and have no hard wired network). The I ran **script** to capture all output.
This is what the output 'typescript' file contains:
```
Script started on Tue Jun  7 18:38:56 2016
bash-3.2$ npm install
bash-3.2$ echo $?
0
```
Note that progress bars were displayed for some initial steps.

However, this output gives the impression that **npm install** succeeded, when it didn't.
